agent/jvm/src/main/java/org/jolokia/jvmagent/JolokiaServer.java

```diff
@@ -412,8 +412,8 @@ public void configure(HttpsParameters params) {
             // get the default parameters
             SSLParameters defaultSSLParameters = context.getDefaultSSLParameters();
 
-            params.setNeedClientAuth(serverConfig.useSslClientAuthentication());
-            defaultSSLParameters.setNeedClientAuth(serverConfig.useSslClientAuthentication());
+            params.setWantClientAuth(serverConfig.useSslClientAuthentication());
+            defaultSSLParameters.setWantClientAuth(serverConfig.useSslClientAuthentication());
```

Comment:
Without adapting the Authenticators later on this would allow people to pass authentication when 

* `useSslClientAuthentication` is true
* No `clientPrincipal` given
* No `extendedClientCheck` set
* No client cert presented.

This shouldn't be the case. Looking into the `ClientCertAuthenticator` its even worse, since the `extendedClientCheck` and `clientPrincipal` verification [is only triggered when a client cert is presented](https://github.com/rhuss/jolokia/blob/69586e7d6321748a27528013b435763bd45d5c03/agent/jvm/src/main/java/org/jolokia/jvmagent/security/ClientCertAuthenticator.java#L87-L88).

But I think you are on the right track here, we just need to verify in `ClientCertAuthenticator` that a valid client certificate, signed by the configured CA (but this hopefully already verified by the SSLEngine then), is presented.